python/ray/actor.py

```diff
@@ -494,357 +494,433 @@ def remote(self, *args, **kwargs):
+                # ActorHandle doesn't need a reference to the ActorMethod.
+                # The ActorMethod has a reference to the ActorHandle and
+                # this was causing cyclic references which were prevent
+                # object deallocation from behaving in a predictable
+                # manner.
+                actor_method_cls = ActorMethod
+                return actor_method_cls(self, attr)
+        except AttributeError:
+            pass
+
+        # If the requested attribute is not a registered method, fall back
+        # to default __getattribute__.
+        return object.__getattribute__(self, attr)
+
+    def __repr__(self):
+        return "Actor(" + self._ray_actor_id.hex() + ")"
+
+    def __del__(self):
+        """Kill the worker that is running this actor."""
+        # TODO(swang): Also clean up forked actor handles.
+        # Kill the worker if this is the original actor handle, created
+        # with Class.remote(). TODO(rkn): Even without passing handles around,
+        # this is not the right policy. the actor should be alive as long as
+        # there are ANY handles in scope in the process that created the actor,
+        # not just the first one.
+        if ray.worker.global_worker.connected and self._ray_original_handle:
+            # TODO(rkn): Should we be passing in the actor cursor as a
+            # dependency here?
+            # self.__ray__terminate__.remote()
+            self._actor_method_call("__ray_terminate__")
+
+    @property
+    def actor_id(self):
+        return self._ray_actor_id
+
+    @property
+    def actor_handle_id(self):
+        return self._ray_actor_handle_id
+
+    def _serialization_helper(self, ray_forking):
+        """This is defined in order to make pickling work.
+
+        Args:
+            ray_forking: True if this is being called because Ray is forking
+                the actor handle and false if it is being called by pickling.
+
+        Returns:
+            A dictionary of the information needed to reconstruct the object.
+        """
+        state = {
+            "actor_id":
+            self._ray_actor_id.id(),
+            "class_name":
+            self._ray_class_name,
+            "actor_forks":
+            self._ray_actor_forks,
+            "actor_cursor":
+            self._ray_actor_cursor.id(),
+            "actor_counter":
+            0,  # Reset the actor counter.
+            "actor_method_names":
+            self._ray_actor_method_names,
+            "actor_method_num_return_vals":
+            self._ray_actor_method_num_return_vals,
+            "method_signatures":
+            self._ray_method_signatures,
+            "actor_creation_dummy_object_id":
+            self._ray_actor_creation_dummy_object_id.id(),
+            "actor_method_cpus":
+            self._ray_actor_method_cpus,
+        }
+
+        if ray_forking:
+            state["forked_actor_handle_id"] = compute_actor_handle_id(
```

Comment:
I'm using this method to compute actor handle IDs for "forked" actor handles. However, if a user pickles and unpickles the actor handle, then I'm generating the actor handle ID randomly.